### DISCUSSION TITLE
Does KaTeX produce MathML Core?

### DISCUSSION THREAD

**Discussion by nschloe** [KaTeX/KaTeX#3893] [Q&A]
Does KaTeX produce MathML Core or another flavor?

**Comment by edemaine** [comment]
KaTeX is designed around the MathML presentation layer. It *might* be MathML Core compatible, but I'm not sure.

**Comment by ronkok** [comment]
KaTeX's MathML code was written before the MathML Core specification existed. If you wish to produce MathML for display, not just for screen readers, then I recommend [Temml](https://temml.org/).

**Reply by nschloe** [reply]
Interesting, never seen it before. What's the difference between a "for display" and "screen reading"?

**Reply by ronkok** [reply]
MathML was originally included in KaTeX for accessibility purposes. By "screen reader", I mean a utility like Microsoft Aria that will read the contents of a web page aloud. Useful for blind users. Such utilities can more easily read KaTeX's MathML output than the HTML output.

By "for display", I just mean the display that you see on your screen. KaTeX has always been very fastidious about the quality of the HTML output and how it looks on the screen. Not so fastidious about how its MathML looks on the screen because its MathML wasn't intended for screen display.

As of January 2023, all three of the major browser engines can display MathML on the screen. But Chromium's MathML is written to a new specification, MathML-Core. In MathML-Core, some things that were previously written as attributes are instead written as CSS rules. The other two browser engines are in a period of (very slow) transition to MathML-Core.

Temml is a fork of KaTeX. I have tried to make Temml emit MathML which will render properly in all three browser engines.
